workflow/controller/test_shared_index_informer_test.go

```diff
@@ -0,0 +1,48 @@
+package controller
+
+import (
+	"time"
+
+	"k8s.io/client-go/tools/cache"
+)
+
+type testSharedIndexInformer struct {
```

Comment:
Could we add this to `workflow/controller/significant_test.go`? I think it would be better to have it all in one file unless it is used in multiple files.